pdata/pcommon/slice.go

```diff
@@ -42,6 +42,7 @@ func NewSlice() Slice {
 // Len returns the number of elements in the slice.
 //
 // Returns "0" for a newly instance created with "NewSlice()".
+// Calling this function on zero-initialized Slice will cause a panic.
```

Comment:
Instead of a comment on all funcs should we have a comment at struct level?

Reply:
So `Value` has this comment: 
https://github.com/open-telemetry/opentelemetry-collector/blob/6b12cc3203488429a3e3d101ee9e9dd152936c6a/pdata/pcommon/value.go#L68-L69

It also has method specific comments specifying when they panic. Happy to update, but when it comes to panics I think it doesn't hurt to have it in both places (I can add it on struct level for `Map`, `Slice`, etc).

Reply:
Given that all the methods in the Slice cause panic while it's not the case for Value, I think it's fine to have it in the Slice's struct comment only.

```
// Must use NewSlice function to create new instances.
// Important: zero-initialized instance is not valid for use, any method call will result in a panic.
```